### PR TITLE
Update readme example to use latest LispObject APIs

### PR DESCRIPTION
**Pull request by Wilfred**
This isn't a perfect example, because we now have `LispObject::as_cons` which provides a safe API. Still, I think it illustrates the principle, and it's better than it was before.

Suggestions for better examples of genuine uses of pointer manipulation are welcome :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -471,15 +471,15 @@ unsafe fn XCAR(object: LispObject) -> LispObject {
 // of XCAR: it only passes valid cons cells. We just use
 // unsafe blocks instead.
 fn car(object: LispObject) -> LispObject {
```
Comment (shaleh) [inline README.md]: We no longer need `unsafe`. Time for a new example!

    if let Some(cons) = object.as_cons() {
        cons.car()
    } else if object.is_nil() {
        LispObject::constant_nil()
    } else {
        wrong_type!(Qlistp, object)
    }

Comment (shaleh) [reply README.md]: There are not many "clean" examples of unsafe. We have wrapped clean APIs on a lot of it. Maybe an example showing access to `globals`? Or an example using the accessor functions to get data out of structs?

Comment (shaleh) [reply README.md]: multibyte.rs has a fair amount of slice creation and what not.

Comment (Wilfred) [reply README.md]: OK, I've updated the example to show dealing with strings, based on our wrapped safe APIs. I wanted to focus on pointers as a fundamental source of unsafety.

**Pull request merged**